src/sentry/tasks/base.py

```diff
@@ -105,10 +107,28 @@ def wrapped(func: Callable[P, R]) -> Task[P, R]:
             wait_for_delivery=wait_for_delivery,
             compression_type=compression_type,
         )(func)
+        if alias or alias_namespace:
+            target_alias = alias if alias else name
+            target_alias_namespace = alias_namespace if alias_namespace else namespace
+            alias_task = target_alias_namespace.register(
+                name=target_alias,
+                retry=retry,
+                expires=expires,
+                processing_deadline_duration=processing_deadline_duration,
+                at_most_once=at_most_once,
+                wait_for_delivery=wait_for_delivery,
+                compression_type=compression_type,
+            )(func)
+            if silo_mode:
+                silo_limiter = TaskSiloLimit(silo_mode)
+                alias_task = silo_limiter(alias_task)
+            target_alias_namespace.set(target_alias, alias_task)
 
         if silo_mode:
             silo_limiter = TaskSiloLimit(silo_mode)
-            return silo_limiter(task)
+            task = silo_limiter(task)
+        namespace.set(name, task)
```

Comment:
I'm assuming you've added `set()` to ensure that the silo wrapped function is the one stored in the task registry.  Perhaps we should apply the silo_limit *before* registering tasks so that we don't need to add a `set()` method?